numba_dppy/compiler.py

```diff
@@ -98,7 +98,7 @@ def compile_with_dppy(pyfunc, return_type, args, debug):
 
     flags = compiler.Flags()
     # Do not compile (generate native code), just lower (to LLVM)
-    if debug:
+    if debug or DEBUGINFO_DEFAULT:
```

Comment:
I don't like that in the middle of pipeline we access some global variable which modifies  behavior.
Much more clean solution would be **to pass correct debug flag to the function**.
On the other hand we have at least 3 decorators, and to pass correct flag we need to modify all of them, but it all comes to this point.
On third hand, our behavior of ```DEGUGINFO``` environment variable is different from Numba behavior. In Numba ```debug``` flag overrides ```DEBUGINFO```. In our case it's vice versa.
@PokhodenkoSA what do you think?

Reply:
Yes, the behavior which is different from numba will confuse users.
Search in numba:
1. There is cool context manager `override_config` for testing config altering, we can create the same for our config
https://github.com/IntelPython/numba/blob/c4f8579f620a38e0d6d6c2e4f5b528899cdcefca/numba/tests/support.py#L515-L515
2. I find 2 places for using `DEBUGINFO_DEFAULT`
https://github.com/IntelPython/numba/blob/0c33c333443971b6d124e3f982238392423a28fb/numba/core/cpu.py#L288-L288
https://github.com/IntelPython/numba/blob/dd57c669e2b8c2dfcdbd2952ea99fe6494b71874/numba/np/ufunc/ufuncbuilder.py#L41-L41
Both are applied for creating default options. In our case it is `compiler.Flags()`. We can inherit from it.
The point is that default `debug` argument value should be `None`. And if it is not None then flag `debuginfo` in `compiler.Flags()` should be forced to it.

Reply:
So `debug` argument should be `None` (i.e. undefined) by default.

Reply:
Setting `debuginfo` to default value inside `compiler.Flags()` sounds like a great idea, if possible

Reply:
> Setting debuginfo to default value inside compiler.Flags()

In separate PR.